satpy/readers/fci_l1c_nc.py

```diff
@@ -686,20 +697,19 @@ def calibrate_rad_to_refl(self, radiance, key):
                 "channel effective solar irradiance set to fill value, "
                 "cannot produce reflectance for {:s}.".format(measured))
             return radiance * np.float32(np.nan)
-
-        sun_earth_distance = np.mean(
-            self.get_and_cache_npxr("state/celestial/earth_sun_distance")) / 149597870.7  # [AU]
-
-        # TODO remove this check when old versions of IDPF test data (<v5) are deprecated.
-        if sun_earth_distance < 0.9 or sun_earth_distance > 1.1:
-            logger.info("The variable state/celestial/earth_sun_distance contains unexpected values"
-                        "(mean value is {} AU). Defaulting to 1 AU for reflectance calculation."
-                        "".format(sun_earth_distance))
-            sun_earth_distance = 1
-
+        sun_earth_distance = self._compute_sun_earth_distance()
         res = 100 * radiance * np.float32(np.pi) * np.float32(sun_earth_distance) ** np.float32(2) / cesi
         return res
 
+    def _compute_sun_earth_distance(self) -> float:
+        """Compute the sun_earth_distance."""
+        if self.is_iqt:
+            sun_earth_distance = 1.0
+            logger.info(f"The value sun_earth_distance is set to {sun_earth_distance}")
```

Comment:
actually, I got a better idea than hardcoding the value to 1 here: could you please implement the sun_earth_distance calculation using these lines instead
https://github.com/pytroll/satpy/blob/3a742c297dfb11ccddaa014b21ecd4ac8eda2c9d/satpy/readers/utils.py#L455-L457
so that when IQT data is used, the corrected reflectance values should not be too different compared to the IDPF data. I would propose to take the time in the middle between observation start and end time for the utc_date.

Please also add a test accordingly, by hardcoding the sun_earth_distance value currently calculated for the time used in the test.